### PR TITLE
Feature, singularity history

### PR DESCRIPTION
**Pull request by ab3327**
Added singularity history tab, it displays Sing time, Sing count,  Quarks this sing, golden quarks this sing, tributes of each cube, total octs ever, c15 exponent and quark hept balance

This also includes awwhy's bug fix for global speed(needed it to get octs in small amounts to not show 0)

### CODE REVIEW

File: src/Reset.ts [hunk 1]
```diff
@@ -900,6 +925,7 @@ export const singularity = async (): Promise<void> => {
     toggleSubTab(9, 0); // set 'corruption main'
     toggleSubTab(-1, 0); // set 'statistics main'
 
+    hold.history.singularity = player.history.singularity;
```
Comment (KhafraDev) [inline src/Reset.ts:928]: ?

File: src/Synergism.ts [hunk 2]
```diff
@@ -1757,7 +1757,7 @@ export const format = (
         return isNaN(input as number) ? '0 [NaN]' : '0 [und.]';
     } else if ( // this case handles numbers less than 1e-6 and greater than 0
         typeof input === 'number' &&
-        input < 1e-12 && // arbitrary number, can be changed
+        input < (!fractional ? 1e-3 : 1e-15) && // arbitrary number, don't change 1e-3
```
Comment (KhafraDev) [inline src/Synergism.ts:1760]: ?

File: src/Synergism.ts [hunk 3]
```diff
@@ -1790,12 +1790,18 @@ export const format = (
     }
 
     // If the power is less than 12 it's effectively 0
-    if (power < -12) {
+    if (power < -15) {
```
Comment (KhafraDev) [inline src/Synergism.ts:1793]: ?

File: src/Synergism.ts [hunk 4]
```diff
@@ -1790,12 +1790,18 @@ export const format = (
...
     }
 
     // If the power is negative, then we will want to address that separately.
     if (power < 0 && !isDecimal(input) && fractional) {
+        if (power <= -15) {
+            return `${format(mantissa, accuracy, long)} / ${Math.pow(10, -power - 15)}Qa`
+        }
+        if (power <= -12) {
+            return `${format(mantissa, accuracy, long)} / ${Math.pow(10, -power - 12)}T`
+        }
```
Comment (KhafraDev) [inline src/Synergism.ts:1804]: ?

**Review by KhafraDev — CHANGES_REQUESTED** [review]

File: src/Synergism.ts (hunk 2 above)
Comment (ab3327) [reply src/Synergism.ts:1760]: this is a live change now, was awwhy's bug fix for global speed not showing negative exponents. without it I had the same bug in the history tab

File: src/Synergism.ts (hunk 3 above)
Comment (ab3327) [reply src/Synergism.ts:1793]: this is a live change now, was awwhy's bug fix for global speed not showing negative exponents. without it I had the same bug in the history tab

File: src/Synergism.ts (hunk 4 above)
Comment (ab3327) [reply src/Synergism.ts:1804]: this is a live change now, was awwhy's bug fix for global speed not showing negative exponents. without it I had the same bug in the history tab

File: src/Reset.ts (hunk 1 above)
Comment (ab3327) [reply src/Reset.ts:928]: need to preserve the sing history over sing otherwise it will reset on sing and you will never see it

**Review by KhafraDev — APPROVED** [review]

**Pull request merged**